cpp/src/arrow/array-test.cc

```diff
@@ -38,7 +36,7 @@ static TypePtr int32 = TypePtr(new Int32Type());
 class TestArray : public ::testing::Test {
  public:
   void SetUp() {
-    pool_ = GetDefaultMemoryPool();
+    pool_ = default_memory_pool();
```

Comment:
https://google.github.io/styleguide/cppguide.html#Function_Names

"Functions that are very cheap to call may instead follow the style for variable names (all lower-case, with underscores between words). The rule of thumb is that such a function should be so cheap that you normally wouldn't bother caching its return value when calling it in a loop. A canonical example is an inline method that just returns one of the class's member variables."

I'm not infallible so if you find any `lower_with_underscores` that may not be cheap to call definitely call me out =)